coordinator/gscoordinator/kubernetes_launcher.py

```diff
@@ -353,13 +425,90 @@ def close_vineyard(self):
         # Use delete deployment instead
         pass
 
-    def create_interactive_instance(self, object_id: int, schema_path: str):
+    def check_if_engine_exist(self, engine_type):
+        """Checks if the engine with the given type exists.
+
+        Args:
+            engine_type: The type of engine to check for.
+
+        Returns:
+            True if the engine exists, False otherwise.
+        """
+
+        engine_pod_name_list = getattr(self, f"_{engine_type}_pod_name_list")
+        engine_pod_ip_list = getattr(self, f"_{engine_type}_pod_ip_list")
+        engine_pod_host_ip_list = getattr(self, f"_{engine_type}_pod_host_ip_list")
+
+        return (
+            engine_pod_name_list is not None
+            and engine_pod_ip_list is not None
+            and engine_pod_host_ip_list is not None
+        )
+
+    def deploy_engine(self, engine_type):
+        """Deploys the engine with the given type.
+
+        Args:
+            engine_type: The type of engine to deploy.
+
+        Returns:
+            A tuple of the pod names, IP addresses, and host IP addresses of the
+            deployed engine.
+        """
+
+        if not self.check_if_engine_exist(engine_type):
+            self._engine_pod_prefix = f"gs-{engine_type}-".replace("_", "-")
+            self._engine_cluster = self._build_engine_cluster(
```

Comment:
If I called deploy_engine multiple times, would I lose the reference to previous cluster? since you set the `self._engine_cluster` repeatedly.